src/model/term/Term.FunctionTerm.js

```diff
@@ -64,6 +64,10 @@ class FunctionTerm extends Term {
   setVariable(from, to){
     this.terms.forEach(term => term.setVariable(from, to));
   }
+
+  getVariables(){
+    return this.terms.map(term => term.getVariables()).flat();
```

Comment:
[`flatMap(… => …)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) is more efficient than `map(… => …).flat()`